### PR TITLE
xds/client: accept resources wrapped in discoverypb.Resource message

### PR DESCRIPTION
**Pull request by menghanl**
RELEASE NOTES: N/A

### CODE REVIEW

File: xds/internal/xdsclient/xdsresource/unmarshal_cds.go [hunk 1]
```diff
@@ -51,6 +51,12 @@ func UnmarshalCluster(opts *UnmarshalOptions) (map[string]ClusterUpdateErrTuple,
 }
 
 func unmarshalClusterResource(r *anypb.Any, f UpdateValidatorFunc, logger *grpclog.PrefixLogger) (string, ClusterUpdate, error) {
+	var err error
+	r, err = unwrapResource(r)
+	if err != nil {
+		return "", ClusterUpdate{}, fmt.Errorf("failed to unwrap resource: %v", err)
+	}
```
Comment (easwars) [inline xds/internal/xdsclient/xdsresource/unmarshal_cds.go:58]: Wouldn't the following work?

```golang
	r, err := unwrapResource(r)
	if err != nil {
		return "", ClusterUpdate{}, fmt.Errorf("failed to unwrap resource: %v", err)
	}
```

And same applies to all calls to `unwrapResource()`

File: xds/internal/xdsclient/xdsresource/unmarshal_eds_test.go [hunk 2]
```diff
@@ -227,6 +228,42 @@ func (s) TestUnmarshalEndpoints(t *testing.T) {
 				Version: testVersion,
 			},
 		},
+		{
+			name:      "v3 endpoints wrapped",
```
Comment (easwars) [inline xds/internal/xdsclient/xdsresource/unmarshal_eds_test.go:232]: Do we need a test for `v2 endpoints wrapped` here like in `unmarshal_cds_test.go`?

File: xds/internal/xdsclient/xdsresource/unmarshal_cds.go (hunk 1 above)
Comment (menghanl) [reply xds/internal/xdsclient/xdsresource/unmarshal_cds.go:58]: Yeah, that works!

File: xds/internal/xdsclient/xdsresource/unmarshal_eds_test.go (hunk 2 above)
Comment (menghanl) [reply xds/internal/xdsclient/xdsresource/unmarshal_eds_test.go:232]: We don't have v2 tests at all...
And this wrapper is hopefully never used in v2, so this should be OK.

**Review by easwars — APPROVED** [review]

**Pull request merged**